### ISSUE TITLE
Show directory tree of possible images

### ISSUE THREAD

**Issue by lamby** [concept-to-clinic/concept-to-clinic#31]
## Overview

The first step of the analysis and identification process is to select an image file from the local disk. We should therefore show the user all the possible files available.

## Expected Behavior

Upon loading the application, we should show a directory tree of the potential files:

![](https://i.imgur.com/nPNtluH.jpg)

## Technical details

The files should be displayed in a tree, ideally, using the Django ``storage`` framework. In the development environement this should be a directory within our project but ignored via a suitable ``.gitignore`` file.

No file format or name filtering is required; we can assume all files in the specified directory are valid. Sorting may need to be applied to the result of Django's ``listdir`` for deterministic ordering.

## Acceptance criteria

All files are listed in their correct hierarchy, sorted alphabetically.

## Blocked by
Issues: #9

**Comment by rahit** [comment]
Hello!
Current API call in index page retrieving image list from `/api/images/`which is using magic method of Django REST frameworks's ModelViewSet to list down all `ImageSeries` object. Since `ImageSeries` is a database model, at first the files needed to be loaded into database to instantiate that way. 

On the other hand, to load files directly from filesystem, we should use Django's `storage` class as you mentioned above. I would like to know which would be the desired system flow? Or we can to use different API endpoint to retrieve files from storage. 

It would be helpful if anybody can give some clarification!

**Comment by lamby** [comment]
@rahit 

Thanks so much for your question. In addition, my sincere apologies for the delay in getting back to you; this notification it stlipped through my "net" so I didn't see it until now..

Ideally, we would want to browse using the ``storage`` framework and — after selection — ``PUT`` the selected files to the API to *create* the ``ImageSeries`` at that point. This ensures a proper separation of the API and this particular frontend. Putting it another way, in this scenario, this ticket is merely about the filesystem browsing step.

However, if we need the ``ImageSeries`` to generate/show the previews, then we will assume that the ``ImageSeries`` already exists. Can you investigate this as a first step? :)

**Comment by rahit** [comment]
Thank you for your response @lamby. According to current set up, it seems the system is assuming that image files would be preloaded in the database. Current vue frontend is expecting `ImageSeries`. API call is also made to an endpoint which expected to return all `ImageSeries` objects. If we want to import image information based on user's selection, I think the better option would be using `ViewSet` instead of `ModelViewSet`. In extended `ViewSet`, we would be able to decouple filesystem API from image series information API.

**Comment by lamby** [comment]
@rahit 

>  the system is assuming that image files would be preloaded in the database

Indeed, and I wonder if the cleanest way forward is to add another endpoint to load images so that all the metadata can be returned by the API instead of the local preview images etc. Let me have a think!

**Comment by rahit** [comment]
@lamby 
I will be right here waiting for you. :musical_note: :)

**Comment by rahit** [comment]
@lamby 
Got it. I will start working on it. But I am not clear what did you mean by -
> Can you open other issue for this

**Comment by lamby** [comment]
@rahit 

I meant that you should preferable add a separate issue that adds another endpoint and commit that separately, then this one "just falls into place". Smaller code changes are, of course, easier to review.

**Comment by lamby** [comment]
Closed as part of #85
